controller/rockcraft.yaml

```diff
@@ -0,0 +1,61 @@
+# Dockerfile https://github.com/kserve/kserve/blob/v0.11.2/Dockerfile
+name: kserve-controller
+summary: KServe controller
+description: "KServe controller manager"
+version: "0.11.2"
+license: Apache-2.0
+base: ubuntu@22.04
+platforms:
+    amd64:
+run-user: _daemon_
+
+services:
+  kserve-controller:
+    override: replace
+    summary: "KServe controller manager service"
+    startup: enabled
+    command: "/manager"
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query --root=${CRAFT_PROJECT_DIR}/../bundles/ubuntu-22.04/rootfs/ -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  controller:
+    plugin: go
+    source: https://github.com/kserve/kserve
+    source-type: git
+    source-tag: v0.11.2
+    build-snaps:
+      - go/1.20/stable
+    build-environment:
+      - CGO_ENABLED: 0
+      - GOOS: linux
+    override-build: |
+
+      # Remove the source from the build directory so we don't
+      # accidentally use the wrong inputs.
+      # The build steps below copy everything that is needed to the working directory
+      rm -rf ./*
+
+      # Copy in the go src
+      cp $CRAFT_PART_SRC/go.mod $CRAFT_PART_SRC/go.sum ./
+
+      go mod download
+
+      cp -r $CRAFT_PART_SRC/cmd ./cmd
+      cp -r $CRAFT_PART_SRC/pkg ./pkg
+
+      # Build
+      go build -a -o manager ./cmd/manager
```

Comment:
If we use the `override-build` are `build-environment` variables used at all in the go build command of this line? Could we just check that is the case?

Reply:
I confirmed that the `build-environment` are used in the `override-build` by echoing the envs from the `override-build`

so the rock looks like:
<details>
  <summary>rockcraft.yaml</summary>

```
# Dockerfile https://github.com/kserve/kserve/blob/v0.11.2/agent.Dockerfile
name: kserve-agent
summary: KServe agent
description: "KServe model agent"
version: "0.11.2"
license: Apache-2.0
base: ubuntu@22.04
platforms:
    amd64:
run-user: _daemon_

services:
  kserve-agent:
    override: replace
    summary: "KServe agent service"
    startup: enabled
    command: "/ko-app/agent"

parts:
  security-team-requirement:
    plugin: nil
    override-build: |
      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
      dpkg-query --root=${CRAFT_PROJECT_DIR}/../bundles/ubuntu-22.04/rootfs/ -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query

  agent:
    plugin: go
    source: https://github.com/kserve/kserve
    source-type: git
    source-tag: v0.11.2
    build-snaps:
      - go/1.20/stable
    build-environment:
      - CGO_ENABLED: 0
      - GOOS: linux
    override-build: |

      echo $CGO_ENABLED
      echo $GOOS

      # Remove the source from the build directory so we don't
      # accidentally use the wrong inputs.
      # The build steps below copy everything that is needed to the working directory
      rm -rf ./*

      # Copy in the go src
      cp $CRAFT_PART_SRC/go.mod $CRAFT_PART_SRC/go.sum ./

      go mod download

      cp -r $CRAFT_PART_SRC/cmd ./cmd
      cp -r $CRAFT_PART_SRC/pkg ./pkg

      # Build
      go build -a -o agent ./cmd/agent

      # Copy the files to the install directory
      cp -r $CRAFT_PART_SRC/third_party/ $CRAFT_PART_INSTALL/third_party/
      mkdir $CRAFT_PART_INSTALL/ko-app
      cp -r agent $CRAFT_PART_INSTALL/ko-app/agent
```

</details>


and by running `rockcraft pack -v`:
<details>
  <summary>packing logs</summary>

```
rockcraft pack -v
Starting rockcraft, version 1.2.3.post17+ga6d85d0                                                                                                                                                  
Logging execution to '/home/ubuntu/.local/state/rockcraft/log/rockcraft-20240319-113554.329146.log'                                                                                                
Launching managed ubuntu 22.04 instance...                                                                                                                                                         
Starting instance                                                                                                                                                                                  
Starting rockcraft, version 1.2.3.post17+ga6d85d0                                                                                                                                                  
Logging execution to '/tmp/rockcraft.log'                                                                                                                                                          
Retrieving base ubuntu@22.04 for amd64                                                                                                                                                             
Retrieved base ubuntu@22.04 for amd64                                                                                                                                                              
Extracting ubuntu:22.04                                                                                                                                                                            
Extracted ubuntu:22.04                                                                                                                                                                             
Initialising lifecycle                                                                                                                                                                             
Installing build-packages                                                                                                                                                                          
Installing build-snaps                                                                                                                                                                             
Skipping pull for agent (already ran)                                                                                                                                                              
Skipping pull for pebble (already ran)                                                                                                                                                             
Skipping pull for security-team-requirement (already ran)                                                                                                                                          
Skipping overlay for agent (already ran)                                                                                                                                                           
Skipping overlay for pebble (already ran)                                                                                                                                                          
Skipping overlay for security-team-requirement (already ran)                                                                                                                                       
Rebuilding agent ('override-build' property changed)                                                                                                                                               
:: + echo 0           # $CGO_ENABLED                                                                                                                                                                             
:: 0                                                                                                                                                                                               
:: + echo linux                    # $GOOS                                                                                                                                                                
:: linux                                                                                                                                                                                           
:: + rm -rf ./CONTRIBUTING.md ./Dockerfile ./LICENSE ./Makefile ./OWNERS ./PROJECT ./README.md ./ROADMAP.md ./SECURITY.md ./agent.Dockerfile ./charts ./cmd ./config ./coverage.sh ./docs ./go.mod ./go.sum ./hack ./install ./pkg ./prow_config.yaml ./python ./qpext ./release ./router.Dockerfile ./test ./third_party ./tools                                                                      
:: + cp /root/parts/agent/src/go.mod /root/parts/agent/src/go.sum ./                                                                                                                               
:: + go mod download                                                                                                                                                                               
:: + cp -r /root/parts/agent/src/cmd ./cmd                                                                                                                                                         
:: + cp -r /root/parts/agent/src/pkg ./pkg                                                                                                                                                         
:: + go build -a -o agent ./cmd/agent                                                                                                                                                              
:: + cp -r /root/parts/agent/src/third_party/ /root/parts/agent/install/third_party/                                                                                                               
:: + mkdir /root/parts/agent/install/ko-app                                                                                                                                                        
:: + cp -r agent /root/parts/agent/install/ko-app/agent                                                                                                                                            
Skipping build for pebble (already ran)                                                                                                                                                            
Skipping build for security-team-requirement (already ran)                                                                                                                                         
Staging agent                                                                                                                                                                                      
Skipping stage for pebble (already ran)                                                                                                                                                            
Skipping stage for security-team-requirement (already ran)                                                                                                                                         
Priming agent                                                                                                                                                                                      
Skipping prime for pebble (already ran)                                                                                                                                                            
Skipping prime for security-team-requirement (already ran)                                                                                                                                         
Packing...                                                                                                                                                                                         
Creating new layer                                                                                                                                                                                 
Created new layer                                                                                                                                                                                  
Creating new user _daemon_                                                                                                                                                                         
Adding user _daemon_:584792 with group _daemon_:584792                                                                                                                                             
Setting the default OCI user to be _daemon_                                                                                                                                                        
Default user set to _daemon_                                                                                                                                                                       
Adding Pebble entrypoint                                                                                                                                                                           
Configuring entrypoint...                                                                                                                                                                          
Entrypoint set to ['/bin/pebble', 'enter', '--verbose']                                                                                                                                            
Configuring Pebble services kserve-agent                                                                                                                                                           
Found 0 Pebble layers in the base's root filesystem                                                                                                                                                
Preparing new Pebble layer file 001-kserve-agent.yaml                                                                                                                                              
Writing new Pebble layer file                                                                                                                                                                      
Adding metadata                                                                                                                                                                                    
Configuring labels and annotations...                                                                                                                                                              
Labels and annotations set to ['org.opencontainers.image.version=0.11.2', 'org.opencontainers.image.title=kserve-agent', 'org.opencontainers.image.ref.name=kserve-agent', 'org.opencontainers.image.licenses=Apache-2.0', 'org.opencontainers.image.created=2024-03-19T11:37:27.348166+00:00', 'org.opencontainers.image.base.digest=722b3bddfe96b95441f626cf94974d79213090ecbd16954f71d7c080fb413561']                                                                                                                                                                                                   
Setting the rock's control data                                                                                                                                                                    
Control data written                                                                                                                                                                               
Metadata added                                                                                                                                                                                     
Exporting to OCI archive                                                                                                                                                                           
Exported to OCI archive 'kserve-agent_0.11.2_amd64.rock'                                                                                                                                           
Packed kserve-agent_0.11.2_amd64.rock
```
</details>
you can see the env vars printed out where I've added a comment